### PR TITLE
remove yield tests from test_lme

### PR DESCRIPTION
**Pull request by jbrockmendel**
This implements a small subset of #3847.

### CODE REVIEW

**Comment by bashtage** [comment]
@josef-pkt LGTM.  Smaller than the big one I put in, but does the job of getting rid of yield.

File: statsmodels/regression/tests/test_lme.py [hunk 1]
```diff
@@ -560,76 +560,94 @@ def test_regularized(self):
...
+# Copied from bashtage's #3847
+@nose.tools.nottest
+@pytest.mark.parametrize('fname,reml,irf',
+                         itertools.product(fnames, [False, True], [False, True]))
+def test_r(fname, reml, irf):
+    ds_ix = int(fname[3:5])
+    do1(reml, irf, ds_ix)
+
+# TODO: Remove after nose is permanently dropped
+def test_r_all():
```
Comment (josef-pkt) [inline statsmodels/regression/tests/test_lme.py]: I guess this will be running the tests a second time when using pytest, does it?

Comment (josef-pkt) [reply statsmodels/regression/tests/test_lme.py]: I guess not if pytests follows the `nottest`

Comment (bashtage) [reply statsmodels/regression/tests/test_lme.py]: pytest respects nottest

**Comment by josef-pkt** [comment]
I think looks fine, not that I tried to check every line

needs rebase, @jbrockmendel I can push a rebase to this branch, unless you want to do the rebase yourself

**Comment by jbrockmendel** [comment]
I'll rebase, want to make clearing these easy ones as frictionless for you as possible.

**Comment by josef-pkt** [comment]
@jbrockmendel it looks like you merged an old master into your branch. That's not a rebase
i.e. full merge commit is (clicking on the commit message in here) https://github.com/statsmodels/statsmodels/commit/ec7898cd6da4c67125e2182c44d00736807e67cb

in the following replace `origin` by whatever alias for statsmodels remote you are using

checkout your branch, then

> git fetch origin
> git rebase remotes/origin/master
then force push, which should be (if default destination is used)
> git push --force

**Comment by jbrockmendel** [comment]
Well there goes my hopes of making this painless for you.  AFAICT you and the pandas maintainer use the term "rebase" differently.  I'll try your more specific instructions, see if I can't remember next time around.

**Comment by bashtage** [comment]
In pandas it should have the same meaning.  My choices:

```
git fetch --all
git rebase upstream/master
# Fix any rebase errors, if any
git push -f
```

**Comment by josef-pkt** [comment]
`rebase` is a git term and command, so there shouldn't be different definitions for it.

**Comment by jbrockmendel** [comment]
> rebase is a git term and command, so there shouldn't be different definitions for it.

I guess just colloquial differences then.  The habit I've gotten into over there is getting a comment from jreback that something needs to be rebased, I run "git pull upstream master", fix any merge conflicts, then "git push".  If "rebase" is not the appropriate term for this process, what is?

**Comment by josef-pkt** [comment]
rebase looks good, currently 2 merges behind master

merging, this should be the last `yield` test fix, AFAICS

@jbrockmendel @bashtage thank you

**Pull request merged**